File: JavaScript/WeightForWeight/weightSort.js
function orderWeight(strng) {
  return strng
          .split(' ')
	  .map(w => { return { number: w, weight: getWeightValue(w) }; })
	  .sort((a,b) => a.weight - b.weight || a.number.localeCompare(b.number))
	  .map(w => w.number)
	  .reduce((r,v) => r + ' ' + v, '').trim();
}

function getWeightValue(num) {
  return num.split('').map(Number).reduce((r,n) => r + n, 0);
}

module.exports = orderWeight;

File: CSharp/CodeWars/WeightForWeight/WeightSort.cs
using System;
using System.Linq;

namespace CodeWars.WeightForWeight
{
    public class WeightSort
    {
        public string OrderWeight(string strng)
        {
            return strng.Split(' ')
                .Select(w => new { Weight = w, Value = GetWeightValue(w) })
                .OrderBy(w => w.Value)
                .ThenBy(w => w.Weight)
                .Select(w => w.Weight)             
                .Aggregate((r, v) => r + ' ' + v)
                .Trim();
        }

        private int GetWeightValue(string weight)
        {  
            return weight.ToCharArray().Select(Char.GetNumericValue).Select(Convert.ToInt32).Sum();
        }
    }
}